File: frameworks-orm/express-orm-naive/src/entities/Comment.ts
import { Entity, PrimaryColumn, Column, ManyToOne } from "typeorm";
import { Post } from "./Post";
import { User } from "./User";

@Entity("tb_comment")
export class Comment {
    @PrimaryColumn()
    id!: string;

    @Column()
    content!: string;

    @Column()
    post_id!: string;

    @Column()
    author_id!: string;

    // NAIVE: Set eager to false to demonstrate N+1 problems
    @ManyToOne(() => Post, post => post.comments, {
        eager: false  // This causes lazy loading!
    })
    post!: Post;

    @ManyToOne(() => User, user => user.comments, {
        eager: false  // This causes lazy loading!
    })
    author!: User;
}

File: frameworks-orm/spring-boot-orm-naive/src/main/java/com/fraiseql/entities/Comment.java
package com.fraiseql.entities;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "tb_comments")
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    // NAIVE: Lazy loading without optimizations
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Post post;

    // NAIVE: Lazy loading without optimizations
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    // Constructors
    public Comment() {}

    public Comment(String content, Post post, User author) {
        this.content = content;
        this.post = post;
        this.author = author;
        this.createdAt = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public Post getPost() { return post; }
    public void setPost(Post post) { this.post = post; }

    public User getAuthor() { return author; }
    public void setAuthor(User author) { this.author = author; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}